diff/parse.go

```diff
@@ -121,12 +121,12 @@ func (r *MultiFileDiffReader) ReadAllFiles() ([]*FileDiff, error) {
 	var ds []*FileDiff
 	for {
 		d, err := r.ReadFile()
-		if d != nil {
-			ds = append(ds, d)
-		}
 		if err == io.EOF {
 			return ds, nil
 		}
+		if d != nil {
+			ds = append(ds, d)
+		}
```

Comment:
I swapped these blocks to preserve the behaviour of ReadAllFiles(), note that none of the tests have changed. The new trailing content is only available when iterating through ReadFile() yourself rather than using this utility function. This is a bit weird, but avoids introducing a parameter to this function, breaking its API.